web/src/components/tooltip.js

```diff
@@ -24,38 +24,46 @@ const Tooltip = ({
 }) => {
   const isMobile = useSelector(state => state.application.isMobile);
 
-  const ref = useRef(null);
-  const width = useWidthObserver(ref);
-  const height = useHeightObserver(ref);
+  const {
+    ref,
+    width,
+    height,
+  } = useRefWidthHeightObserver();
 
   const screenWidth = window.innerWidth;
   const screenHeight = window.innerHeight;
 
   if (!position) return null;
 
+  // Note: at first render, width and height will be undefined
+  // They will only be set once the DOM node has been created
+  const hasDimensions = width && height;
+
   const style = {};
-  let x = 0;
-  let y = 0;
-  // Check that tooltip is not larger than screen
-  // and that it does fit on one of the sides
-  if (2 * MARGIN + width >= screenWidth
-    || (position.x + width + MARGIN >= screenWidth && position.x - width - MARGIN <= 0)) {
-    // TODO(olc): Once the tooltip has taken 100% width, it's width will always be 100%
-    // as we base our decision to revert based on the current width
-    style.width = '100%';
-  } else {
-    x = position.x + MARGIN;
-    // Check that tooltip does not go over the right bound
-    if (width + x >= screenWidth) {
-      // Put it on the left side
-      x = position.x - width - MARGIN;
+  if (hasDimensions) {
```

Comment:
I wrapped all the resizing logic inside this wrapper in order to make sure this doesn't happen before it should. This helped a bit with performance and fixed some weird layout bugs I could see at some point